src/hotspot/share/cds/filemap.cpp

```diff
@@ -245,30 +268,37 @@ void FileMapHeader::populate(FileMapInfo* mapinfo, size_t core_region_alignment)
   _requested_base_address = (char*)SharedBaseAddress;
   _mapped_base_address = (char*)SharedBaseAddress;
   _allow_archiving_with_java_agent = AllowArchivingWithJavaAgent;
-  // the following 2 fields will be set in write_header for dynamic archive header
-  _base_archive_name_size = 0;
-  _base_archive_is_default = false;
 
   if (!DynamicDumpSharedSpaces) {
-    set_shared_path_table(mapinfo->_shared_path_table);
+    set_shared_path_table(info->_shared_path_table);
     CDS_JAVA_HEAP_ONLY(_heap_obj_roots = CompressedOops::encode(HeapShared::roots());)
   }
 }
 
+// Do not check _magic, it's not been set yet.
```

Comment:
I think you can get rid of this comment by moving line 228 .. 231 after line 233.